frictionless/checks/regulation.py

```diff
@@ -59,6 +59,80 @@ def validate_row(self, row):
     }
 
 
+class table_dimensions(Check):
+    """Check for minimum and maximum table dimensions
+
+    API      | Usage
+    -------- | --------
+    Public   | `from frictionless import checks`
+    Implicit | `validate(checks=[{"code": "table-dimensions", min_rows, max_rows, min_fields, max_fields}])`
+
+    Parameters:
+       descriptor (dict): check's descriptor
+
+    """
+
+    code = "table-dimensions"
+    Errors = [errors.TableDimensionsError]
+
+    def __init__(
+        self, descriptor=None, *, min_rows=-1, max_rows=-1, min_fields=-1, max_fields=-1
+    ):
+        self.setinitial("min_rows", min_rows)
+        self.setinitial("max_rows", max_rows)
+        self.setinitial("min_fields", min_fields)
+        self.setinitial("max_fields", max_fields)
+        super().__init__(descriptor)
+        self.__min_rows = self["min_rows"]
+        self.__max_rows = self["max_rows"]
+        self.__min_fields = self["min_fields"]
+        self.__max_fields = self["max_fields"]
+
+    # Validate
+
+    def validate_start(self):
+        number_fields = len(self.resource.schema.fields)
+        if number_fields < self.__min_fields:
+            yield errors.TableDimensionsError(
+                note="Current number of fields is %s, the minimum is %s"
+                % (number_fields, self.__min_fields)
+            )
+        if number_fields > self.__max_fields:
+            yield errors.TableDimensionsError(
+                note="Current number of fields is %s, the maximum is %s"
+                % (number_fields, self.__max_fields)
+            )
+
+    def validate_row(self, row):
+        self.__last_row = row
+        if self.__max_rows > 0 and self.__last_row.row_number > self.__max_rows:
```

Comment:
- it might be better to move this condition to `validate_end` so it's run only once not for every row